internal/bufpushaction/bufpushaction.go

```diff
@@ -0,0 +1,313 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufpushaction
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/bufbuild/buf-push-action/internal/pkg/github"
+	"github.com/bufbuild/buf/private/pkg/app/appcmd"
+	"github.com/bufbuild/buf/private/pkg/app/appflag"
+	"github.com/bufbuild/buf/private/pkg/storage"
+	"github.com/bufbuild/buf/private/pkg/storage/storageos"
+	"github.com/spf13/cobra"
+	"go.uber.org/multierr"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	bufTokenKey         = "BUF_TOKEN"
+	githubTokenKey      = "GITHUB_TOKEN"
+	githubRepositoryKey = "GITHUB_REPOSITORY"
+)
+
+// Main is the entrypoint to the buf CLI.
+func Main(name string) {
+	appcmd.Main(context.Background(), NewRootCommand(name))
+}
+
+func NewRootCommand(name string) *appcmd.Command {
+	builder := appflag.NewBuilder(name, appflag.BuilderWithTimeout(120*time.Second))
+	return &appcmd.Command{
+		Use:   name,
+		Short: "helper for the GitHub Action buf-push-action",
+		SubCommands: []*appcmd.Command{
+			{
+				Use:   "push <input> <track> <git-commit-hash>",
+				Short: "push to BSR",
+				Args:  cobra.ExactArgs(3),
+				Run:   builder.NewRunFunc(runPush, interceptErrorForGithubAction),
+			},
+			{
+				Use:   "delete-track <input> <track>",
+				Short: "delete a track on BSR",
+				Args:  cobra.ExactArgs(2),
+				Run:   builder.NewRunFunc(runDeleteTrack, interceptErrorForGithubAction),
+			},
+		},
+	}
+}
+
+func runPush(ctx context.Context, container appflag.Container) error {
+	input := container.Arg(0)
+	track := container.Arg(1)
+	currentGitCommit := container.Arg(2)
+
+	// exit early if buf or BUF_TOKEN isn't found
+	if _, err := exec.LookPath("buf"); err != nil {
+		return errors.New(`buf is not installed; please add the "bufbuild/buf-setup-action" step to your job found at https://github.com/bufbuild/buf-setup-action`)
+	}
+	if container.Env(bufTokenKey) == "" {
```

Comment:
```suggestion
	if _, err := exec.LookPath("buf"); err != nil {
		return errors.New(`buf is not installed; please add the "bufbuild/buf-setup-action" step to your job found at https://github.com/bufbuild/buf-setup-action`)
	}
	// exit early if buf or BUF_TOKEN isn't found
	if container.Env(bufTokenKey) == "" {
```
or just delete this comment altogether

Reply:
deleted